fhir-notification-nats/README.MD

```diff
@@ -0,0 +1,89 @@
+        "clientId": "fhir-server",
+        "servers": "nats://nats-node1:4222,nats://nats-node2:4222,nats://nats-node3:4222",
+        "useTLS": false,
+        "truststoreLocation": "resources/security/nats.client.truststore.jks",
+        "truststorePassword": "change-password",
+        "keystoreLocation": "resources/security/nats.client.keystore.jks",
+        "keystorePassword": "change-password"
+    }
+ 
+Set "enabled" to true and provide the name of your NATS cluster for the value of "cluster".  You can leave "channel" and "clientId" as defined.  Provide the URL for one or more NATS servers as the value for "servers". See the "Use TLS" section below for a discussion of the TLS settings, if required.  
+
+For use with the test configuration in the next section, set "enabled" to true and make no other changes.
+
+## Create a simple NATS cluster for testing
+
+In the directory fhir-server-test/src/test/resources/nats you will find the files to set up a basic NATS cluster and IBM FHIR server on the same docker network.  Follow these steps to test:
+
+* Build your IBM FHIR image with the NATS configuration discussed above and publish it to Docker Hub.  Edit the fhir_server service image in fhir-server-test/src/test/resources/nats/docker-compose.yml to point to this new image.  
+
+	```    
+fhir_server:
+		image: <your-repo>/<your-image>
+```
+Note: As an alternative to building and publishing your own image, you can start an IBM FHIR server image that contains the NATS publisher functionality, then copy the updated configuration file containing the above changes into the IBM FHIR container and restart the container.  
+ 
+	```
+	Example: 
+docker cp ./fhir-server-config.json fhir_server:/opt/ol/wlp/usr/servers/fhir-server/config/default/fhir-server-config.json
+```        
+
+* Start the NATS cluster and IBM FHIR server.  
+
+	```
+cd fhir-server-test/src/test/resources/nats
+docker-compose up -d
+```  
+You can view the logs with:  
+
+	```
+docker-compose logs -f
+```
+
+* Start the NATS subscriber using [Node.js](https://nodejs.org/en/download/).  
+
+	```
+npm install node-nats-streaming
+node nats-subscriber
+```
+* POST a patient record to IBM FHIR server and observe the NATS message received by the subscriber in the console output.
+
+## Use TLS (advanced)
+
+To use TLS (which you must do if your NATS cluster requires TLS) set "useTLS" to true in fhir-server-config.json and provide client truststore and keystore locations and passwords as the remaining config values.  Furthermore, to use TLS to connect to a NATS cluster, that NATS cluster must be enabled to use TLS.  The example above does not use TLS.
+
+### To create a truststore and keystore
+
+On MacOS, install the [mkcert tool](https://github.com/FiloSottile/mkcert) with `brew install mkcert`, then follow these steps to create a Java keystore and truststore for use with NATS.  You will also need Java keytool and openssl installed. Set $PASSWORD to be any value. 
```

Comment:
```suggestion
On MacOS, install the [mkcert tool](https://github.com/FiloSottile/mkcert) with `brew install mkcert`, then follow these steps to create a Java keystore and truststore for use with NATS.  You also need Java keytool and openssl installed. Set $PASSWORD to be any value. 
```
I don't see any advantage to using this approach.  I think pointing to the openssl commands is best. 

Typical pattern in the docs is not to use $PASSWORD rather use change-password and tell the users to change for their environment.

Reply:
We don't have those openssl commands available now and we'll need to track those down and re-test.  We'll target that for a future update.  $PASSWORD changed to change-password.